src/Autodesk.Forge.Core/HttpResponseMessageExtensions.cs

```diff
@@ -38,7 +38,7 @@ public static async Task<HttpResponseMessage> EnsureSuccessStatusCodeAsync(this
                 {
                     errorMessage = $"\nMore error details:\n{errorMessage}.";
                 }
-                throw new HttpRequestException($"The server returned the non-success status code {(int)msg.StatusCode} ({msg.ReasonPhrase}).{errorMessage}");
+                throw new HttpRequestException($"The server returned the non-success status code {(int)msg.StatusCode} ({msg.ReasonPhrase}).{errorMessage}", null, msg.StatusCode);
             }
```

Comment:
fix the issue reported here: https://github.com/Autodesk-Forge/forge-api-dotnet-core/issues/70#issue-890954746

Reply:
@szilvaa-adsk there is another requirement: https://github.com/Autodesk-Forge/forge-api-dotnet-core/issues/70#issuecomment-841099809
Should we fix it? and How to do it?

Reply:
I think we should derive a new `TooManyRequestsException` from `HttpRequestException` and throw that specifically. This new exception could then have a `TimeSpan RetryAfter` property with the value from the header.